### PR TITLE
Widget 2.0 - just to rebase

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers the plan is to merge this into the mobile pr, then merge that down into the base one. Rebase master so i can get the correct tab-bar, then fixup mobile and the widget 2.0. 
Widget 2.0 needs the dropdown menu and some small fixups.
so... treat this as a WIP, point out anything that seems crazy, but we'll get a second pass before it goes in

### CODE REVIEW

File: shared/folders/list.desktop.js [hunk 1]
```diff
@@ -54,26 +55,33 @@ class Render extends Component<void, Props, State> {
...
-          <Row
-            key={rowKey(t.users)}
-            {...t}
-            isPublic={this.props.isPublic}
-            ignored={false}
-            isFirst={!idx} />
-          ))}
-        <Ignored ignored={this.props.ignored} showIgnored={this.state.showIgnored} styles={styles}
-          isPublic={this.props.isPublic} onToggle={() => this.setState({showIgnored: !this.state.showIgnored})} />
+      <Box style={{...stylesScrollContainer, ...this.props.style}}>
```
Comment (chrisnojima) [inline shared/folders/list.desktop.js]: so we can scroll the list in the widget

**Comment by chrisnojima** [comment]
Note: i went down the path of completely sharing the component. added the concept of a smallMode and made the favorite rpc convert to the dumb data we want. it currently doesn't tell us when people are broken but we'll need that later anyways. marking everyone good for now

File: shared/menubar/index.js [hunk 2]
```diff
@@ -151,37 +149,37 @@ class Menubar extends Component {
...
+    // const {username} = this.props
+    // const folders = (this.props.folders || []).map((f: Folder) : FolderInfo => { // eslint-disable-line arrow-parens
+      // return {
+        // type: 'folder',
+        // folderName: f.name,
+        // isPublic: !f.private,
+        // // TODO we don't get this information right now,
+        // isEmpty: false,
+        // openFolder: () => {
+          // setImmediate(() => this.props.openInKBFS(`${getTLF(!f.private, f.name)}`))
```
Comment (chrisnojima) [inline shared/menubar/index.js]: all this stuff needs to get re-hooked up. i'll clean it up when we merge down

File: shared/menubar/index.render.js.flow [hunk 3]
```diff
@@ -1,35 +1,37 @@
...
+export default class Render extends Component<void, Props, void> {}
 
-export type FolderInfo = {
-    type: 'folder',
-    folderName: string,
-    isPublic: boolean,
-    isEmpty: boolean,
-    openFolder: () => void
-}
+// export type FolderInfo = {
```
Comment (chrisnojima) [inline shared/menubar/index.render.js.flow]: i'll clean this up too, just wanted it as a ref until i sync it all up

File: shared/reducers/favorite.js [hunk 4]
```diff
@@ -2,18 +2,44 @@
...
 type State = {
-  folders: ?Array<Folder>
+  folders: ?Props
 }
 
 const initialState = {
   folders: null
 }
 
+const folderToProps = (folders, username) => {
```
Comment (chrisnojima) [inline shared/reducers/favorite.js]: this could exist in the action also, either way

File: shared/folders/render.desktop.js [hunk 5]
```diff
@@ -24,13 +24,27 @@ class Render extends Component<void, Props, State> {
     return (
       <Box style={stylesContainer}>
         <TabBar tabWidth={130}>
-          <TabBar.Item label={'private/' + this.props.privateBadge} selected={this.state.showPrivate}
-            onClick={() => this.setState({showPrivate: true})}>
-            <List {...this.props.private} />
+          <TabBar.Item label={'private/' + (this.props.privateBadge ? this.props.privateBadge : '')} selected={this.state.showPrivate}
```
Comment (MarcoPolo) [inline shared/folders/render.desktop.js]: 🐑 this.props.privateBadge || ''

**Comment by chrisnojima** [comment]
@MarcoPolo fixed

**Comment by chrisnojima** [comment]
i'm going ot merge this down since it'll be PRed again later

**Pull request merged**